opensight_restserver.py

```diff
@@ -58,6 +59,42 @@ def connect_to_tcp(host, port):
     return client
 
 
+def recv_timeout(the_socket, timeout=2):
+    #make socket non blocking
+    the_socket.setblocking(0)
+    
+    #total data partwise in an array
+    total_data=[]
+    data=''
+    
+    #beginning time
+    begin=time.time()
+    while 1:
+        #if you got some data, then break after timeout
+        if total_data and time.time()-begin > timeout:
+            break
+        
+        #if you got no data at all, wait a little longer, twice the timeout
+        elif time.time()-begin > timeout*2:
```

Comment:
Are you try to do something like exponential backoff? If so then should add `timeout = timeout*2`
If not then just `*2` might not make much difference if `timeout == 0.05` then the total timeout is only 0.1. A slow network might cause an infinite loop.